storage/sas_token.go

```diff
@@ -82,10 +82,15 @@ func ParseAccountSASConnectionString(connString string) (map[string]string, erro
 	for _, atoken := range tokens {
 		// The individual k-v are separated by an equals sign.
 		kv := strings.SplitN(atoken, "=", 2)
+		if len(kv) != 2 {
```

Comment:
can we add a test for this use-case?

Reply:
Not sure it will fix it - the problem is the SAS key uses, as most ofd the key formats, "=" for padding. It's URL encoded (as %3D) when generated so maybe the problem is caused by you decoding it while it should be kept as it is. Example:

`?sv=2018-03-28&ss=bfqt&srt=sco&sp=rwdlacup&se=2018-12-07T20:01:48Z&st=2018-12-07T12:01:48Z&spr=https&sig=JZ83Ie1N%2BqgqNMlM1Y%2FCw0p0pDQ8A1%2B7G%2FlCVwVpHHM%3D`

Reply:
This was primarily to prevent the panic, and shed light on to what was causing it.

The `SplitN` should also only split on the 1st `=`s?